### PR TITLE
Firestore connections

### PR DESCRIPTION
**Pull request by dgutoryan**

### CODE REVIEW

File: src/app/auth/auth.module.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+@NgModule({
+  imports: [
+    CommonModule,
+    HttpClientModule,
+    ReactiveFormsModule,
+    AngularFireAuthModule,
+  ],
+  declarations: [LoginComponent],
+  providers: [
+    LoginService
```
Comment (AndreyLukashenko) [inline src/app/auth/auth.module.ts]: 😑

File: src/app/auth/auth.module.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  ],
+  exports: [LoginComponent]
```
Comment (AndreyLukashenko) [inline src/app/auth/auth.module.ts]: 😑

File: src/app/auth/login/login.service.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  constructor(private firebaseAuth: AngularFireAuth,) {
+    this.user = firebaseAuth.authState;
+   }
+
+  login({ email, password }) {
+    this.firebaseAuth
+      .auth
+      .signInWithEmailAndPassword(email, password)
+      .then(data => {
+        localStorage['user'] = JSON.stringify(data.user);
```
Comment (AndreyLukashenko) [inline src/app/auth/login/login.service.ts]: It's more appropriate using `localStorage.setItem`

File: src/app/auth/login/login.service.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+      });
+  }
+
+  logout() {
+    this.firebaseAuth.auth.signOut();
+    localStorage['user'] = '';
```
Comment (AndreyLukashenko) [inline src/app/auth/login/login.service.ts]: And the same like above

File: src/app/auth/login/login.service.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+import { AngularFireAuth } from 'angularfire2/auth';
+import * as firebase from 'firebase';
+
+@Injectable({
+  providedIn: 'root'
```
Comment (AndreyLukashenko) [inline src/app/auth/login/login.service.ts]: 😑

File: src/app/auth/login/login.service.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+})
+export class LoginService {
+  user: Observable<firebase.User>;
+
+  constructor(private firebaseAuth: AngularFireAuth,) {
+    this.user = firebaseAuth.authState;
+   }
+
+  login({ email, password }) {
```
Comment (AndreyLukashenko) [inline src/app/auth/login/login.service.ts]: Oh distructuring, you're good 👍🏾

File: src/app/auth/login/login.service.ts [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+import { AngularFireAuth } from 'angularfire2/auth';
+import * as firebase from 'firebase';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class LoginService {
+  user: Observable<firebase.User>;
+
+  constructor(private firebaseAuth: AngularFireAuth,) {
```
Comment (AndreyLukashenko) [inline src/app/auth/login/login.service.ts]: It seems to me that there's unnecessary comma here

File: src/app/auth/login/login.component.ts [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+  constructor(
+    private loginService: LoginService,
+  ) { }
+
+  ngOnInit() {
+
+  }
+
+  login() {
+    this.loginService.login(this.loginForm.value)
```
Comment (AndreyLukashenko) [inline src/app/auth/login/login.component.ts]: 😑

File: src/app/auth/auth.module.ts (hunk 1 above)
Comment (dgutoryan) [reply src/app/auth/auth.module.ts]: Done

File: src/app/auth/auth.module.ts (hunk 2 above)
Comment (dgutoryan) [reply src/app/auth/auth.module.ts]: Done

File: src/app/auth/login/login.service.ts (hunk 3 above)
Comment (dgutoryan) [reply src/app/auth/login/login.service.ts]: Done

File: src/app/auth/login/login.service.ts (hunk 4 above)
Comment (dgutoryan) [reply src/app/auth/login/login.service.ts]: Done

File: src/app/auth/login/login.service.ts (hunk 5 above)
Comment (dgutoryan) [reply src/app/auth/login/login.service.ts]: Done

File: src/app/auth/login/login.service.ts (hunk 7 above)
Comment (dgutoryan) [reply src/app/auth/login/login.service.ts]: Done

File: src/app/auth/login/login.component.ts (hunk 8 above)
Comment (dgutoryan) [reply src/app/auth/login/login.component.ts]: Done

**Pull request merged**